contracts/domainsSetLibrary.sol

```diff
@@ -0,0 +1,284 @@
+// SPDX-License-Identifier: MIT
+
+pragma solidity ^0.8.18;
+
+
+library domainSet {
+    struct Domain {
+        string name;
+        address owner;
+        uint256 deposit;
+    }
+
+    struct Set {
+        Domain[] _values;
+        mapping(string => uint256) _positions;
+    }
+
+    function _add(
```

Comment:
There is no restriction to register domain if its parent domain does not exist.   -1 point